### PR TITLE
tests

### PR DESCRIPTION
**Pull request by gr2m**
- removes existing test setup
- refactors code to make it test-able
- README: add issue form to example and fix example issue body and expected output
- Add test for the example shown in the README

### CODE REVIEW

**Comment by gr2m** [comment]
This is pretty much the test setup that I'm currently using for my actions.

I'm using your action quite a bit, I'll just keep adding stuff to make it more convenient to maintain my fork, such as an automated release setup.

I'm happy to help you get the automated release setup work for your action, but if it's getting too much I am happy to just maintain my fork

**Comment by stefanbuck** [comment]
This is awesome. Will take a closer look later today. Meanwhile I'm going to add you as a collaborator.

File: index.js [hunk 1]
```diff
@@ -1,92 +1,131 @@
...
+    `${env.HOME}/issue-parser-result.json`,
+    jsonStringify(json),
+    "utf-8"
+  );
+  core.setOutput("jsonString", jsonStringify(json));
 }
 
+// We wrap the code in a `run` function to enable testing.
+// On GitHub Actions the `run` function is executed immediately.
+// `NODE_ENV` is set when running tests on GitHub Actions as part of CI.
```
Comment (stefanbuck) [inline index.js:124]: Nice setup 😍

File: README.md [hunk 2]
```diff
@@ -34,20 +81,21 @@ A bug happened!
...
-    "browsers": "chrome, safari",
-    "never_give_up": true,
-    "hot_dog_is_a_sandwich": false
+  "contact": "me@me.com",
+  "what_happened": "A bug happened!",
+  "version": "1.0.0",
+  "browsers": "Chrome, Safari",
+  "never_give_up": true,
+  "hot_dog_is_a_sandwich": false
 }
```
[the remark this answers is not in the document]
Comment (stefanbuck) [reply README.md:100]: I'm going to raise a separate PR for this.

**Review by stefanbuck — APPROVED** [review]
This is amazing. Thanks also for the readme changes, much better. Thank you so much @gr2m

**Pull request merged**